dali/operators/image/resize/resize_attr.cc

```diff
@@ -0,0 +1,318 @@
+        // and use it for the missing dimensions;
+        double avg_scale = 1;
+        for (int d = 0; d < spatial_ndim_; d++) {
+          scale[d] = out_size[d] / in_size[d];
+          if (mask[d])
+            avg_scale *= std::abs(scale[d]);  // abs because of possible flipping
+        }
+        if (sizes_provided > 1)
+          avg_scale = std::pow(avg_scale, 1.0 / sizes_provided);
+        for (int d = 0; d < spatial_ndim_; d++) {
+          if (!mask[d]) {
+            scale[d] = avg_scale;
+            out_size[d] = in_size[d] * scale[d];
+          }
+        }
+      }
+      if (has_max_size_) {
+        for (int d = 0; d < spatial_ndim_; d++) {
+          if (max_size_[d] > 0 && std::abs(out_size[d]) > max_size_[d]) {
+            out_size[d] = std::copysignf(max_size_[d], out_size[d]);
+            scale[d] = out_size[d] / in_size[d];
+          }
+        }
+      }
+    } else {
+      // NotLarger or NotSmaller mode
+
+      // First, go through all the dimensions that have their scale defined and find
+      // the min (or max) scale
+      double final_scale = 0;
+      bool first = true;
+      for (int d = 0; d < spatial_ndim_; d++) {
+        if (mask[d]) {
+          float s = std::abs(scale[d]);
+          if (first ||
+              (mode_ == ResizeMode::NotSmaller && s > final_scale) ||
+              (mode_ == ResizeMode::NotLarger && s < final_scale))
+            final_scale = s;
+          first = false;
+        }
+      }
+
+      // If there's a size limit, apply it and possibly reduce the final scale
+      if (has_max_size_) {
+        for (int d = 0; d < spatial_ndim_; d++) {
+          if (max_size_[d] > 0) {
+            double s = static_cast<double>(max_size_[d]) / in_size[d];
+            if (s < final_scale)
+              final_scale = s;
+          }
+        }
+      }
+
+      // Now, let's aply the final scale to all dimensions - if final_scale is different (in
+      // absolute value) than one defined by the caller, adjust it; also, if no scale was defined
+      // for a dimension, then just use the final scale.
+      for (int d = 0; d < spatial_ndim_; d++) {
+        if (!mask[d]) {
+          out_size[d] = in_size[d] * final_scale;
+          scale[d] = final_scale;
+        } else if (std::abs(scale[d]) != final_scale) {
+          scale[d] = std::copysign(final_scale, scale[d]);
+          out_size[d] = in_size[d] * scale[d];
+        }
+      }
+    }
+  }
+}
+
+void ResizeAttr::CalculateSampleParams(ResizeParams &params,
+                                       SmallVector<float, 3> requested_size,
+                                       SmallVector<float, 3> input_size) {
+  assert(static_cast<int>(requested_size.size()) == spatial_ndim_);
+  assert(static_cast<int>(input_size.size()) == spatial_ndim_);
+
+
+  for (int d = 0; d < spatial_ndim_; d++) {
+    DALI_ENFORCE(input_size[d] != 0 || requested_size[d] == 0,
+                "Cannot produce non-empty output from empty input");
```

Comment:
Does it make any sense to just pad here?

Reply:
Resampling kernel can't pad.